pkg/eventmonitor/consumers/process.go

```diff
@@ -0,0 +1,171 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux || windows
+
+package consumers
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"github.com/DataDog/datadog-agent/pkg/eventmonitor"
+	"github.com/DataDog/datadog-agent/pkg/security/secl/model"
+)
+
+// ProcessCallback is a callback function that will be called when a process exec/exit event is received
+// Defined as an alias to avoid type incompatibilities
+type ProcessCallback = func(uint32)
+
+// ProcessConsumer represents a consumer of process exec/exit events that can be subscribed to
```

Comment:
what about fork events? some consumers might be interested in fork events, maybe provide a parameter to decide?

Reply:
For now I think neither DI nor GPU are interested in fork events, and I didn't want to add too many things to this to keep it simple.

Reply:
this is a infrastructure package. I don't think it should contain specific code for DI or GPU. if you build a generic consumer let's make it properly, otherwise let's move it to a different location

Reply:
Added support to fork events, also added an options struct to avoid having too many unnamed arguments in the constructor.